va/va_cp.h

```diff
@@ -0,0 +1,420 @@
+ * as defined by the H.264 specification.
+ *
+ */
+typedef struct _VACencSliceParameterBufferH264 {
+
+    /** \brief Parameters from \c nal_unit() of the slice.*/
+    /**@{*/
+    /** \brief  Same as the H.264 bitstream syntax element. */
+    uint8_t nal_ref_idc;
+    /** \brief Indicate if this is coded slice of an IDR picture. 
+     * Corresponds to IdrPicFlag as the H.264 specification.*/
+    uint8_t idr_pic_flag;
+    /**@}*/
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t slice_type;
+    /** \brief Indicate if this is a field or frame picture.
+     * \c VA_FRAME_PICTURE, \c VA_TOP_FIELD, \c VA_BOTTOM_FIELD*/
+    uint8_t field_frame_flag;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint32_t frame_number;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint32_t idr_pic_id;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t pic_order_cnt_lsb;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t delta_pic_order_cnt_bottom;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t delta_pic_order_cnt[2];
+    /**
+     * \brief decoded reference picture marking. Information for \c 
+     * dec_ref_pic_marking() as defined by the H.264 specification.
+     */
+    /**@{*/
+    union {
+        struct {
+            /** \brief Same as the H.264 bitstream syntax element. */
+            uint32_t no_output_of_prior_pics_flag             : 1;
+            /** \brief Same as the H.264 bitstream syntax element. */
+            uint32_t long_term_reference_flag                 : 1;
+            /** \brief Same as the H.264 bitstream syntax element. */
+            uint32_t adaptive_ref_pic_marking_mode_flag       : 1;
+            /** \brief number of decode reference picture marking. */
+            uint32_t dec_ref_pic_marking_count                : 8;;
+            /** \brief Reserved for future use, must be zero */
+            uint32_t reserved                                 : 21;
+            } bits;
+        uint32_t value;
+    } ref_pic_fields;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t memory_management_control_operation[32];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t difference_of_pic_nums_minus1[32];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t long_term_pic_num[32];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t max_long_term_frame_idx_plus1[32];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t long_term_frame_idx[32];
+    /**@}*/
+    /** \brief Pointer to the next #VACencSliceParameterBufferH264 element,
+    * or \c NULL if there is none.*/
+    void *next;
+    /** \brief Reserved bytes for future use, must be zero */
+    unsigned long va_reserved[VA_PADDING_MEDIUM];
+} VACencSliceParameterBufferH264;
```

Comment:
Can you explain how providing the reference pictures to the decoder works here?  With the current structures I'm not sure you have enough information to fill RefPicList0/RefPicList1.